src/stdlib_child_list.f90

```diff
@@ -0,0 +1,369 @@
+
+        !Reduce the count by 1
+        this_child_list%num_nodes = this_child_list%num_nodes - 1
+    end subroutine pop_node_at_tail
+
+    !> Removing the node at the given 'node_index' from the input child list
+    pure subroutine remove_node_at_index( this_child_list, node_index )
+
+        class(child_list), intent(inout) :: this_child_list
+        integer, intent(in):: node_index
+        type(node), pointer:: current_node
+        
+        ! This index will be reference for child list
+        integer:: index
+
+        !iterating through the child_list to reach the nth node
+        current_node => this_child_list%head
+        
+        ! return if the given node index is not in range of 1 to size of linked list
+        if(node_index<=0) return;
+        if(node_index>this_child_list%num_nodes) return;
+        index = 1
+        do while ( associated(current_node) )
+            if (index==node_index) then
+            if (associated(current_node%prev).and.associated(current_node%next)) then
+                !child_list Node is in mid
+                current_node%next%prev => current_node%prev
+                current_node%prev%next => current_node%next
+
+            else if (associated(current_node%prev)) then
+                !child_list tail
+                nullify(current_node%prev%next)
+                this_child_list%tail => current_node%prev
+
+            else if (associated(current_node%next)) then
+                !child_list head
+                nullify(current_node%next%prev)
+                this_child_list%head => current_node%next
+            else
+                !only node in list
+                nullify(this_child_list%head)
+                nullify(this_child_list%tail)  
+            end if
+
+            !Destroy node content and Free it's memory
+            call current_node%clear()
+            deallocate(current_node)
+
+            !Reduce the index by 1
+            this_child_list%num_nodes = this_child_list%num_nodes - 1
+            return
+            end if
+            current_node => current_node%next
+            index = index+1
+        end do
+    end subroutine remove_node_at_index
+
+
+    !> Returns the pointer to the item stored at 'node_index' in the input child list
+    !> 
+    !> Returns a pointer
+    function get_node_at_index( this_child_list, node_index ) result (return_item)
+
+        class(child_list), intent(inout) :: this_child_list
+        integer, intent(in):: node_index
+        class(*), pointer :: return_item
+        type(node), pointer:: current_node
+        integer:: index
+        
+        !iterating through the child_list to reach the nth node
+        current_node => this_child_list%head
+        index = 1
+        do while ( associated(current_node) )
+        
+            if (index == node_index) then
+                ! Return the pointer to item stored at specified index
+                return_item => current_node%item
```

Comment:
Maybe a routine can be abstracted here, allowing the user to customize the way to get the contents of the linked list?
```fortran
...
generic :: get => get_node_at_index, &
                  get_node_at_index_user
procedure, private :: get_node_at_index, &
                      get_node_at_index_user
...
abstract interface
        subroutine get_value(this_item, return_item)
            class(*), intent(in) :: this_item
            class(*), intent(out) :: return_item
        end subroutine get_value
end interface
...
subroutine get_node_at_index_user(this_list, node_index, func, return_item)
        class(child_list), intent(inout) :: this_list
        integer, intent(in) :: node_index
        procedure(get_value) :: func
        class(*), intent(out) :: return_item
        type(node), pointer :: curr_node
        integer index

        curr_node => this_list%head
        index = 1
        do while (associated(curr_node))
            if (index == node_index) then
                call func(curr_node%item, return_item)
                nullify (curr_node)
                return
            end if

            curr_node => curr_node%next
            index = index + 1
        end do
        nullify (curr_node)

end subroutine get_node_at_index_user
...
```

The user can use it like this:
```fortran
program main

    use linked_list_m, only: child_list, rk, ik
    implicit none
    real(rk) :: x
    integer(ik) :: n

    type(this_child) :: list
    print *, list%size()

    call list%push(1.0_rk)
    call list%push(2_ik)

    call list%get(1, get_value, x)
    call list%get(2, get_value, n)

    print *, x, n, list%size()

contains

    subroutine get_value(this_item, return_item)
        class(*), intent(in) :: this_item
        class(*), intent(out) :: return_item

        select type (this_item)
        type is (real(rk))
            select type (return_item)
            type is (real(rk))
                return_item = this_item
            class default
                print *, '*<ERROR>* get_value: type mismatch'
            end select
        type is (integer(ik))
            select type (return_item)
            type is (integer(ik))
                return_item = this_item
            class default
                print *, '*<ERROR>* get_value: type mismatch'
            end select
        class default
            print *, "*<ERROR>* get_value: invalid type"
        end select

    end subroutine get_value

end program main
```